diracx-routers/src/diracx/routers/__init__.py

```diff
@@ -440,3 +446,46 @@ async def db_transaction(db: T2) -> AsyncGenerator[T2]:
         if reason := await is_db_unavailable(db):
             raise DBUnavailable(reason)
         yield db
+
+
+class ClientMinVersionCheckMiddleware(BaseHTTPMiddleware):
+    """Custom FastAPI middleware to verify that
+    the client has the required minimum version.
+    """
+
+    def __init__(self, app: FastAPI):
+        super().__init__(app)
+        self.min_client_version = get_min_client_version()
+
+    async def dispatch(self, request: Request, call_next) -> Response:
+        client_version = request.headers.get("DiracX-Client-Version")
+        if client_version and self.is_version_too_old(client_version):
+            # When comes from Swagger or Web, there is no client version header.
+            # This is not managed here.
+            raise HTTPException(
+                status_code=HTTPStatus.UPGRADE_REQUIRED,
+                detail=f"Client version ({client_version}) not recent enough (>= {self.min_client_version}). Upgrade.",
+            )
+
+        response = await call_next(request)
+        return response
+
+    def is_version_too_old(self, client_version: str) -> bool | None:
+        """Verify that client version is ge than min."""
+        return parse(client_version) < parse(self.min_client_version)
```

Comment:
Actually if you don't catch the `InvalidVersion` exception, FastAPI returns a `500` error to the user, which is not what you want actually.

What about?

```suggestion
    def is_version_too_old(self, client_version: str) -> bool:
        """Verify that client version is ge than min."""
        try:
            return parse(client_version) < self.parsed_min_client_version)
        except InvalidVersion:
            raise HTTPException(
                status_code=400,
                detail=f"Invalid version string: '{version_str}'"
            )
```

Note1: the return type is `bool` in my example, because the function raises an `HTTPException` if there is an issue with the version.

Note2: `parse(self.min_client_version)` is replaced by `self.parsed_min_client_version` for 2 reasons. (i) even though the `parse` function is cheap, it is executed on every request, so it is just a very minor optimization; (ii) if for some reason there is an issue with `min_client_version` (e.g. typo like `1.2]4`), it would raise a `500` error, which is correct in this case, because it would come from the server configuration, and thus we avoid any ambiguity.